wgpu_macros/src/generic_bindings.rs

```diff
@@ -980,7 +980,7 @@ pub fn sub_module_generic_bindings(input: TokenStream) -> TokenStream {
 
     let input_params = process_params(input_vec);
 
-    let mut rng = rand::thread_rng();
+    let mut rng = StdRng::seed_from_u64(0);
```

Comment:
So this is debatable.

I've changed `rand::thread_rng()` to `seed_from_u64(0)`, which comes with the benefit of types actually being deterministic at compile time. I.e., if you recompile the same program multiple times you will actually get a context of the same type. I think this would be pretty helpful for debugging. And it's just not as weird as not having deterministic compilation? 

The one downside is now someone could theoretically do something like `ContextXX::new()`, but I don't think that's possible in reality. I tried doing this and it gave me a bunch of type inference issues due to the generics that are part of this, which I think would be very difficult to avoid, so I think that's fine.

Reply:
What if we feature gated this instead? Default non-determinism with a feature that allows determinism for debugging. I personally don't see how this helps debugging but if it can be then there should be that option. I'm a little worried it could get abused though.

Reply:
I mean just in general the idea of non-deterministic compilation feels absurd to me. I don't even really care about the debugging aspect that much, I was just trying to throw that in hoping it would sweeten the deal for you. People here are trying to make completely reproducible builds and you're making them non-deterministic and calling it a feature lmao. Just calling `rand` at all at compilation is crazy to me.

Anyway, I don't see how it could get abused if the `new` function is marked as `unsafe`.

Reply:
You still can't really use it as a function argument as the generic types are hard to get right. This doesn't change any functionality other than making compilation non-deterministic.

Reply:
With this change, what happens if I want to create two different contexts in the same program? For instance, in trivial_pipe.rs or the shadowmap example?

Reply:
Otherwise this breaks some examples

Reply:
Damn it, for some reason I had thought that this function would provide a new random for each call of the function, even if it was deterministic. But apparently not. I guess my point is moot, then.